File: service/nodejs/send.js
import amqp from 'amqplib/callback_api'
import { QUEUE_NAME, RABBITMQ_URL } from './secret'

amqp.connect(RABBITMQ_URL, function (err, connection) {
    connection.createChannel(function (errCon, channel) {
        const queue = QUEUE_NAME
        const msg = {
            username: '',
            title: '',
            message: '',
        }
        const jsonMessage = JSON.stringify(msg)
        channel.assertQueue(queue, { durable: false })
        channel.sendToQueue(queue, Buffer.from(jsonMessage))
        console.log('[*] Sent %s', msg)
        setTimeout(function () {
            connection.close()
            process.exit(0)
        }, 1000)
    })
})

File: service/python/send.py
import pika
import secret
import json


notification = {
    'username': '',
    'title': '',
    'message': '',
    'deviceToken': secret.deviceToken
}

notification = json.dumps(notification)

params = pika.URLParameters(secret.rabbitmqUrl)
connection = pika.BlockingConnection(params)
channel = connection.channel()

channel.queue_declare(queue=secret.queueName, passive=False,
                      durable=False, auto_delete=False)

channel.basic_publish(
    exchange='', routing_key=secret.queueName, body=notification)
print('[*] Sent: ' + notification)
connection.close()
